ceph/ceph_admin/bootstrap.py

```diff
@@ -52,22 +55,124 @@ def construct_registry(cls, registry: str, json_file: bool = False):
     return config_dict_to_string(reg_args)
 
 
-class BootstrapMixin:
-    """Add bootstrap support to the child class."""
+def validate_fsid(cls, fsid: str, out: str) -> bool:
+    """
+    Method is used to validate fsid
+    Args:
+        cls: class object
+        fsid: <ID>
+        out: bootstrap response logs
 
-    def get_method(self, key):
-        """
-        Fetch method using the Key
-        Args:
-            key: fsid
+    Returns: Boolean value whether fsid is valid
+
+    """
+    out, err = cls.shell(args=["ceph", "fsid"])
+    return out == fsid
 
-        Returns: Returns the method based on the Key
 
-        """
-        verify_map = {
-            "fsid": self.validate_fsid,
+def validate_dashboard(cls, out, user=None, password=None):
+    """
+    Method to validate dashboard login using provided user/password
+    and Also validates user/passwd in bootstrap console output
+
+    Args:
+        cls: class object
+        out: bootstrap response log
+        user: dashboard user
+        password: dashbard password
+
+    Returns:
+        boolean
+    """
+    _, _, db_log_part = out.partition("Ceph Dashboard is now available at:")
+
+    if not db_log_part:
+        return False
+
+    def parse(string):
+        str_ = r"%s:\s(.*)" % string
```

Comment:
```suggestion
        str_ = r"%s:\s(.+)" % string
```

`*` captures zero or more characters, and `+` captures one or more characters, and we always expect to find one or more here.

Reply:
done